Source/Diagnostics/ReducedDiags/ChargeOnEB.cpp

```diff
@@ -190,6 +190,9 @@ void ChargeOnEB::ComputeDiags (const int step)
                 // Given that only a tiny fraction of the cells have a non-zero contribution
                 // (the ones that intersect with the EB), it is not clear whether ReduceOpSum
                 // or AtomicAdd would be faster. However, the implementation with AtomicAdd is easier.
+#ifdef AMREX_USE_OMP
+#pragma omp critical
+#endif
                 amrex::Gpu::Atomic::AddNoRet( surface_integral_pointer, local_integral_contribution );
```

Comment:
It would be more efficient to use `amrex::HostDevice::Atomic::Add(urface_integral_pointer, local_integral_contribution)` instead of omp critical.